scipy/optimize/tests/test_zeros.py

```diff
@@ -302,6 +315,69 @@ def test_root_scalar_fail(self):
         with pytest.raises(ValueError):
             root_scalar(f1, method='halley', fprime2=f1_2, x0=3, xtol=1e-6)  # no fprime
 
+    def test_array_chandrupatla(self):
+        """test _chandrupatla with array, similar to newton method"""
+
+        k = np.arange(1,8)
+        result = zeros._chandrupatla(self.f3, 0, 3, args=(k,))
+        x_expected = ([1., 1.41421356, 1.73205081,
+                      2., 2.23606798,2.44948974,
+                      2.64575131])
+        assert_allclose(result.x, x_expected)
+
+    def test_chandrupatla_newton_similar(self):
+        """test that _chandrupatla is giving a similar result to newton"""
+
+        k = np.arange(1,8)
+
+        result = zeros._chandrupatla(self.f3, 0, 3, args=(k[0],))
+        expected = zeros.newton(self.f3, x0=0, x1=3, args=(k[0],))
+        assert_allclose(result.x, expected)
+
+    def test_chandrupatla_brent_similar(self):
+        """test than chandrupatla gives equal scalar result to brentq"""
+        def f3(x):
+            return (x**3 - 1)
+
+        result = zeros._chandrupatla(f3,0,3,maxiter=50)
+        expected = zeros.brentq(f3,0,3)
+
+        assert_allclose(result.x, expected)
+
```

Comment:
```suggestion
```

This only tests that `_chandrupatla` can solve a scalar equation. There's so much more to test than that, and it will be quite clear from all the other tests that `_chandrupatla` can do that, so I don't think this is needed.